core/flavours/generic.go

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) 2017 Cisco and/or its affiliates.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package flavours
+
+import (
+	"github.com/ligato/cn-infra/core"
+	"github.com/ligato/cn-infra/db/keyval/etcdv3"
+	"github.com/ligato/cn-infra/logging/logrus"
+	"github.com/ligato/cn-infra/messaging/kafka"
+	"github.com/ligato/cn-infra/utils/config"
+	"github.com/namsral/flag"
+)
+
+// Generic is set of common used generic plugins. This flavour can be used as a base
+// for different flavours. The plugins are initialized in the same order as they appear
+// in the structure.
+type Generic struct {
+	etcdConfigFile  string
+	kafkaConfigFile string
+
+	Lg    *logrus.Plugin
+	Etcd  *etcdv3.Plugin
+	Kafka *kafka.Plugin
+}
+
+// RegisterFlags registers the options that need to be parsed.
+func (f *Generic) RegisterFlags() {
+	flag.StringVar(&f.etcdConfigFile, "etcdv3-config", "", "Location of the Etcd configuration file; also set via 'ETCDV3_CONFIG' env variable.")
+	flag.StringVar(&f.kafkaConfigFile, "kafka-config", "", "Location of the Kafka configuration file; also set via 'KAFKA_CONFIG' env variable.")
+}
+
+// ApplyConfig loads the config and creates the plugins.
+func (f *Generic) ApplyConfig() error {
```

Comment:
We need to simplify the way how you choose plugins in the flavor:
- there needs to be default dir for configs
- there needs to be default config file name
- in flavor you just pick plugins you need + inject them
- we need to avoid using flags in very flavor
- let's use the the ConfigBroker in a particular plugin (and just override the config file name if needed)